### PR TITLE
Adds a starter docs page on Azure

### LINKED ISSUE #561: Does tf-controller support workload identity for Azure?
Does Terraform Controller  support workload identity for Azure?(see https://github.com/Azure/azure-workload-identity)

After helmrelease the terraform controller to Azure, when I deploy the Terraform resource, I get the following authentication error in tf-runner pod.
Appropriate Azure Managed ID privileges are granted to the workload identity for the serviceaccount, so if Terraform Controller supports it, an authentication error should not occur, but is it not currently supported?

```
❯ kubectl get pod tf-runner -o yaml
    message: |+
      error running Init: rpc error: code = Internal desc = exit status 1

      Error: Failed to get existing workspaces: Error retrieving keys for Storage Account "<storageaccount>": azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.azure.com/subscriptions/<subscriptionid>/resourceGroups/<resourceGroups>/providers/Microsoft.Storage/storageAccounts/<storageaccount>/listKeys?api-version=2021-01-01: StatusCode=404 -- Original Error: adal: Refresh request failed. Status Code = '404'. Response body: getting assigned identities for pod <namespace>/tf-runner in CREATED state failed after 16 attempts, retry duration [5]s, error: <nil>. Check MIC pod logs for identity assignment errors
       Endpoint http://xxx.xxx.xxx.xxx/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fmanagement.azure.com%2F
```

### PR DESCRIPTION
**Pull request by LappleApple**
Closes #561, with content provided by the users. This is just a starter page for the topic to close the issue.

### CODE REVIEW

File: docs/use_tf_controller/with-azure.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+```
+       - name: ARM_USE_OIDC
+          value: "true"
+        - name: ARM_OIDC_TOKEN_FILE_PATH
+          value: "/var/run/secrets/azure/tokens/azure-identity-token"
+```
+
+Example yaml:
+
+```
```
Comment (yitsushi) [inline docs/use_tf_controller/with-azure.md:18]:
```suggestion
```yaml
```
for syntax highlight

File: docs/use_tf_controller/with-azure.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+Use the OIDC flag and explicitly point to the token. Due to a bug in AzureRM 3.44.x, use version 3.47.x or later.
+
+Set env variables on the runner pod:
+
+```
+       - name: ARM_USE_OIDC
+          value: "true"
+        - name: ARM_OIDC_TOKEN_FILE_PATH
+          value: "/var/run/secrets/azure/tokens/azure-identity-token"
```
Comment (yitsushi) [inline docs/use_tf_controller/with-azure.md:13]:
```suggestion
        - name: ARM_USE_OIDC
          value: "true"
        - name: ARM_OIDC_TOKEN_FILE_PATH
          value: "/var/run/secrets/azure/tokens/azure-identity-token"
```
indentation

File: docs/use_tf_controller/with-azure.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          value: ""
+        - name: ARM_TENANT_ID
+          value: ""
+        - name: ARM_CLIENT_ID
+          value: ""
+        - name: ARM_OIDC_TOKEN_FILE_PATH
+          value: "/var/run/secrets/azure/tokens/azure-identity-token"
+```
+
+Import existing resources to some tfstate stored on a storage account.
```
Comment (yitsushi) [inline docs/use_tf_controller/with-azure.md:61]: > to some tfstate stored

that feels a bit off, aybe drop the `some` ?

**Review by yitsushi — COMMENTED** [review]
lgtm, some quick fix suggestions (indentation issue, add syntax highlight)

File: docs/use_tf_controller/with-azure.md (hunk 3 above)
Comment (LappleApple) [reply docs/use_tf_controller/with-azure.md]: @yitsushi Good catch, thank you -- I made an adjustment

**Review by yitsushi — APPROVED** [review]

**Pull request merged**